core/src/banking_stage/unprocessed_transaction_storage.rs

```diff
@@ -736,10 +769,20 @@ impl ThreadLocalUnprocessedPackets {
     /// Checks sanitized transactions against bank, returns valid transaction indexes
     fn filter_invalid_transactions(
         transactions: &[SanitizedTransaction],
+        invalid_fee_payer_filter: &InvalidFeePayerFilter,
         bank: &Bank,
         total_dropped_packets: &mut usize,
     ) -> Vec<usize> {
-        let filter = vec![Ok(()); transactions.len()];
+        let filter = transactions
+            .iter()
+            .map(|tx| {
+                if invalid_fee_payer_filter.should_reject(&tx.message().account_keys()[0]) {
+                    Err(TransactionError::InsufficientFundsForFee)
```

Comment:
I was about to suggest to use a new TransactionError specifically to indicate it's failed because the payer account is black listed.

Mainly thinking from UX perspective; sophisticated abusers will soon find way to get around this filter, the 2 sec blackout would probably apply more to innocent mistakes, a clear error might help in that case?

Reply:
Unrecorded transaction errors do not filter up to the users though. It'd purely be an internal metric - similar to our `AccountInUse` error metrics.

Reply:
@t-nelson the actual errors here don't get used at all. We only check `is_ok()` below.
But I changed agree `InvalidAccountForFee` is a better dummy error here since we've effectively marked the account as invalid with the filter: d52a238bbb